backend/packages/Upgrade/src/api/controllers/SegmentController.ts

```diff
@@ -246,18 +249,17 @@ export class SegmentController {
    *          description: Internal Server Error, SegmentId is not valid
    */
   @Get('/:segmentId')
-  public getSegmentById(@Param('segmentId') segmentId: string, @Req() request: AppRequest): Promise<Segment> {
-    if (!segmentId) {
-      return Promise.reject(new Error(SERVER_ERROR.MISSING_PARAMS + ' : segmentId should not be null.'));
+  public async getSegmentById(@Param('segmentId') segmentId: string, @Req() request: AppRequest): Promise<Segment> {
```

Comment:
This is still returning a 500 in the HTTP response. We should validate the param with an id validator (see the implementation for feature flags).